pkg/kubelet/util/manager/watch_based_manager.go

```diff
@@ -208,14 +266,35 @@ func (c *objectCache) Get(namespace, name string) (runtime.Object, error) {
 		// we limit ourselves to just quickly stop the reflector here.
 		if c.isImmutable(object) {
 			if item.stop() {
+				item.markImmutable()
 				klog.V(4).Infof("Stopped watching for changes of %q/%q - object is immutable", namespace, name)
 			}
 		}
+		item.updateLastAcquiredTime(c.clock.Now())
 		return object, nil
 	}
 	return nil, fmt.Errorf("unexpected object type: %v", obj)
 }
 
+func (c *objectCache) startPeriodicRecycle() {
+	if c.maxIdleTime == 0 {
+		return
+	}
+
+	go wait.Forever(func() {
```

Comment:
I don't think this PR works.
The fact the secrets whenever changed re automatically updated to all pods mounting them:
https://kubernetes.io/docs/concepts/configuration/secret/#mounted-secrets-are-updated-automatically
Means that they are periodically checked if nothing changed.


So this PR doesn't work.

Reply:
The following uses configmap as an example. The secret is the same as the configmap,
The configmap is used in the following two places:
1. Used by a container as volume.
2. Information in the configmap is injected to the container as environment variables.

For the configmap volume, we expect that the watch link can be maintained. Therefore, we set maxIdleTime to three times the value of resyncInterval. (Each resyncInterval period, VolumeManager queries the configmap information, lastAcquiredTime will be updated.)

For the configmap used by environment variables, it is expected that after the container is created, the kubelet can disconnect the watch link for the configmap.

Reply:
OK - I see what you're saying now - that makes sense.
(we can argue if three times resyncInterval isn't too small - I would feel more comfortable with something higher, but that's a minor thing)

But this kind of change for sure deserves a test.

Reply:
If a ConfigMap or Secret is only used to set container environment variables, why watch it at all?